engines/cegar_values.cpp

```diff
@@ -105,13 +107,29 @@ class ValueAbstractor : public smt::IdentityWalker
           return Walker_Continue;
         }
 
+        fresh_solver_->push();
```

Comment:
this seems like a hack to check if a value is positive or negative. Right? If so, please add a comment to improve this in future.

Reply:
Yeah added with 1d86cbb257beb95ab3b712739dc17bc20773633d